echopop/live/live_biology.py

```diff
@@ -0,0 +1,1377 @@
+    # -------- Initialize the dataframe
+    aged_unaged_sex_proportions = aged_sex_df.reset_index().set_index(
+        ["species_id", "sex"] + spatial_column
+    )
+    # --------- Add the within-unaged weight proportions
+    aged_unaged_sex_proportions["weight_proportion_unaged"] = length_weight_sex_proportions.stack()
+    # --------- Add the overall-unaged weight proportions
+    aged_unaged_sex_proportions["weight_proportion_overall_unaged"] = (
+        unaged_weight_sex_proportions_overall.stack()
+    )
+    # ---- Overall aged and unaged proportions
+    aged_unaged_proportions = aged_proportions.reset_index(name="aged_proportions")
+    # ---- Set index
+    aged_unaged_proportions.set_index(spatial_column + ["species_id"], inplace=True)
+    # -------- Add unaged proportions
+    aged_unaged_proportions["unaged_proportions"] = unaged_proportions  # .reset_index()
+    # ---- Reset the index
+    aged_unaged_proportions = aged_unaged_proportions.reset_index()
+
+    # Return output
+    return {
+        "aged_weight_proportions_df": aged_overall_df,
+        "unaged_weight_proportions_df": average_length_bin_weights_df,
+        "aged_unaged_sex_weight_proportions_df": (
+            aged_unaged_sex_proportions.astype(float).reset_index().fillna(0.0)
+        ),
+        "aged_unaged_weight_proportions_df": aged_unaged_proportions,
+    }
+
+
+# TODO: NEED TO UPDATE TO EITHER INSERT IF NOT PRESENT OR UPDATE OTHERWISE ! ! !
+# ! SEE ABOVE
+def summarize_strata(
+    nasc_biology_data: pd.DataFrame, spatial_data: pd.DataFrame, file_configuration: dict
+):
+
+    # Get biology database
+    acoustic_db = file_configuration["database"]["acoustics"]
+
+    # Get biology database
+    biology_db = file_configuration["database"]["biology"]
+
+    # Validate table
+    if not SQL(biology_db, "validate", table_name="strata_summary_df"):
+
+        # Create copy
+        strata_df = spatial_data.copy()
+
+        # Define new columns
+        strata_df[
+            [
+                "length_mean",
+                "weight_mean",
+                "TS_mean",
+                "number_density_mean",
+                "biomass_density_mean",
+                "abundance_sum",
+                "biomass_sum",
+            ]
+        ] = np.nan
+        # ---- Drop 'latitude_interval'
+        strata_df.drop(columns=["latitude_interval"], inplace=True)
+
+        # ---- Create
+        SQL(
+            biology_db,
+            "create",
+            table_name="strata_summary_df",
+            dataframe=strata_df,
+            primary_keys=["stratum"],
+        )
+        # ---- Populate table
+        SQL(
+            biology_db,
+            "insert",
+            table_name="strata_summary_df",
+            dataframe=strata_df,
+            id_columns=["stratum"],
+        )
+
+    # Get unique strata values
+    strata_values = np.unique(nasc_biology_data["stratum"]).tolist()
+
+    # Update the table
+    sql_update_strata_summary(
```

Comment:
This is called in both `biology_pipeline` and `acoustic_pipeline`. Not sure why this update from `acoustic_db` to `biology_db` needs to happen?